src/services/helpers.ts

```diff
@@ -23,14 +23,27 @@ export class NetworkError extends Error {
   }
 }
 
+const timeoutAfter = (seconds: number): Promise<Error> => {
```

Comment:
just curious, was looking [around](https://stackoverflow.com/questions/42147733/doing-a-timeout-error-with-fetch-react-native/54208009) and `AbortController` was talked about as a recommended way to abort `fetch`,  maybe you can share why you decided to use `Promise.race` and a custom timeout instead?

Reply:
abortController() aborts the connections and Promise.race returns fetch data or throw specify error after x seconds which ever that is earlier. 

updated the code to abort connection properly after timeout as recommended.